docs/Developers/NextGen_Core.rst

```diff
@@ -0,0 +1,286 @@
+NextGen-Core
+============
+
+This document provides a brief overview of coala's NextGen-Core.
+coala's NextGen-Core comes with the promise of resolving the dependencies
+amongst bears more efficiently (by removing the queuing mechanism) and better
+performance
+
+What is new?
+------------
+
+The following new features have been added as a part of the NextGen-Core:
+
+- Easier Interface
+- Official support for virtual files
+- Improved dependency system
+- New base bear type: ``DependencyBear``
+- Ability to modify bear dependencies at runtime
+- Custom parallelization for bears
+- Superior caching
+
+What has changed?
+-----------------
+
+- ``Global Bears`` are now called ``Project Bears`` and ``Local Bears`` are
+  now known as ``File Bears``. Both the ``ProjectBear`` and the ``FileBear``
+  classes inherit from the base class ``coalib.core.Bear``.
+- There is no need to pass ``HiddenResults`` between different bears (made
+  possible by the new dependency management system) to hide results to the
+  result callback. Only the results that were explicitly requested by passing
+  the needed bears are passed now. There is no need to pass instances of
+  ``Result`` anymore, any python object will do.
```

Comment:
Nah that's not quite, you still have to pass `Result`s, but it's possible to supply arbitrary objects if needed. This is only useful for dependency bears. To communicate issues to coala you have to use `Result` instances still.